File: android/src/main/java/com/supermap/rnsupermap/JSEncodeType.java
package com.supermap.rnsupermap;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Myself on 2017/8/16.
 */

public class JSEncodeType extends ReactContextBaseJavaModule {
    public static final String REACT_CLASS="JSEncodeType";

    public JSEncodeType(ReactApplicationContext context){
        super(context);
    }

    @Override
    public String getName(){return REACT_CLASS;}

    @Override
    public Map<String, Object> getConstants() {
        final Map<String, Object> constants = new HashMap<>();
        constants.put("NONE", 0);
        constants.put("BYTE", 1);
        constants.put("INT16", 2);
        constants.put("INT24", 3);
        constants.put("INT32", 4);
        constants.put("DCT", 8);
        constants.put("SGL", 9);
        constants.put("LZW", 11);
        return constants;
    }
}

File: ios/oldNative/JSEncodeType.m
//
//  JSEncodeType.m
//  Supermap
//
//  Created by 王子豪 on 2017/8/16.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import "JSEncodeType.h"
#import "SuperMap/EncodeType.h"
@implementation JSEncodeType
    RCT_EXPORT_MODULE();
-(NSDictionary*)constantsToExport{
    return @{@"NONE":@(NONE),
             @"BYTE":@(BYTE),
             @"INT16":@(INT16),
             @"INT24":@(INT24),
             @"INT32":@(INT32),
             @"DCT":@(DCT),
             @"SGL":@(SGL),
             @"LZW":@(LZW)};
}
@end
